### PR TITLE
Spotify Attempted Integration

### PR DESCRIPTION
**Pull request by YourEpicness**
This is multiple attempts at trying to get integration with Spotify to work. Spotify API is pretty outdated so the documentation is hard to follow. One thing I got working was getting pure client-side authentication working aka implicit grant: https://developer.spotify.com/documentation/general/guides/authorization/implicit-grant/.

I also attempted to get server-side authorization to work by using the routes/spotify.router and adding authorization logic to controllers/spotify.controllers. I haven't tested it, but it most likely doesn't work due to different packages for making requests. I've also tried using the node-fetch library for making requests but end up with "Can't find .js modules, please convert to .mjs" errors. For request libraries: https://www.npmjs.com/package/request (deprecated) vs https://www.npmjs.com/package/axios (modern) vs https://www.npmjs.com/package/node-fetch (lightweight-modern)
Documentation for previous attempt: https://developer.spotify.com/documentation/general/guides/authorization/code-flow/

For the third attempt, I tried to get the Spotify Web Playback SDK to work. I was able to get it to function properly, but it only allows for you to select the "browser" as a device to listen to Spotify from. Documentation for that is here https://developer.spotify.com/documentation/web-playback-sdk/

If someone else wants to pursue this or can get this to work or already knows, please go ahead.

TLDR: Multiple attempts, client-side works (barely), server-side doesn't. A bunch of links to documentation. If you want to try and work from where I left off go ahead.

### CODE REVIEW

**Comment by YourEpicness** [comment]
[Review pull request for Spotify](https://trello.com/c/OHlf38qz/14-review-pull-request-for-spotify)

**Review by yeiltza — APPROVED** [review]
Ok, I see what you where talking about. I'll meet with you to figure this out.
